src/Microsoft.PowerShell.Commands.Utility/commands/utility/trace/MshHostTraceListener.cs

```diff
@@ -104,6 +104,7 @@ public override void WriteLine(string output)
             try
             {
                 _cachedWrite.Append(output);
+                _cachedWrite.Insert(0, DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.ffff "));
```

Comment:
Can we append (not insert) before `output`?

Reply:
I considered that originally, but there is a `Write()` method above this one that can be used to construct the `_cachedWrite` prior to `WriteLine()` being called.  I want to make sure the timestamp is left most.

Reply:
I'd expect that trace is as fast as possible. Here call `Now()` that is slow. Then `Insert()` - again slow.
Maybe add the timestamp in `WriteDebugLine()` implementation called below?

Reply:
`WriteDebugLine()` is used for all `DEBUG:` output though and I don't think it makes sense to have a timestamp there all the time

Reply:
My only concern is that we will slow down a script execution too much. Also we will allocate tons Datetime() objects.

We could use Environment.TickCount https://source.dot.net/#System.Private.CoreLib/src/System/Environment.CoreCLR.cs,102
This also corresponds to "determine where the command/expression is spending its time"

Oops, it seems ':' is a separator in the trace output but Datetime has the char in time part too - it can break trace file parsers.

Reply:
TickCount is ok, but less readable or obvious what that number represents to someone seeing it the first time.  This ONLY affects use of `trace-command -pshost` so it's opt-in for both tracing and sending output to the host.  It doesn't make sense to parse this output as it would normally send the traces to logging and not the host.

Reply:
This output can be redirected to a file. It seems I look at it from the other side. :-)

It is not blocking comment.